src/db/AnalyticsRepository.ts

```diff
@@ -16,33 +16,29 @@ class AnalyticsRepository {
     // Execute a query for an endpoint.
     async execute(endpoint: string, sqlVars = {}): Promise<any> {
         // Look up the query text for the endpoint
-        const query = this.queries.get(endpoint)
+        let query = this.queries.get(endpoint)
 
         if (!query) {
             throw new Error(`Query not found for endpoint ${endpoint}`)
         }
 
-        // Iterate over the values and escape them
-        const setStatements = []
+        // Originally based on the variable system in MySQL
+        // as the optimizer had huge problems with @VARS in queries
+        // we changed to use our own replacement system
+
+        // escape first and then replace all occurences of the key in the query with the escaped value
         for (const [key, value] of Object.entries(sqlVars)) {
             const escapedValue: string =
                 // bit weird, but this is the only way to get the escape function without a type error
                 (this.pool as unknown as Connection).escape(value)
-            setStatements.push(`SET @${key}=${escapedValue};`)
+            query = query?.replaceAll(`@${key}`, escapedValue)
         }
 
+        console.log(query)
+
         // Execute the query and return the result
-        const [rows, _] = await this.pool.query(
-            `${setStatements.join(' ')} ${query}`
-        )
-
-        // as we send multiple queries, we need to return only the last one
-        // which contains the actual result
-        if (setStatements.length > 0 && Array.isArray(rows)) {
-            return rows[rows.length - 1]
-        } else {
-            return rows
-        }
+        const [rows, _] = await this.pool.query(query)
+        return rows
```

Comment:
Does this handle multiple queries?

Reply:
the code for multiple queries was removed but you still need that construct.